src/heap.h

```diff
@@ -5,28 +5,114 @@ template<class ValueType>
 class Heap
 {
 private:
-    int heapSize;
     Node<ValueType>* root;
+    int size;
+    int height;
 
-protected:
+    int calculateHeapHeight(int size);
+    Node<ValueType>* insertValue(ValueType value, Node<ValueType>* node, int level);
+    void rebuild(Node<ValueType>* node);
     void swapNodes(Node<ValueType>* node1, Node<ValueType>* node2);
+    void deleteHeap(Node<ValueType>* node);
+
+protected:
+    bool virtual compare(Node<ValueType>* node, Node<ValueType>* toCompare) = 0;
 
 public:
     Heap();
-    virtual ~Heap() = default;
+    virtual ~Heap();
 
     void show();
-    void virtual insertValue(ValueType value) = 0;
+    void insertValue(ValueType value);
     void virtual build(ValueType values[]) = 0;
     ValueType virtual extract() = 0;
     ValueType virtual get() = 0;
 };
 
 template<class ValueType>
-Heap<ValueType>::Heap() : heapSize(0), root(nullptr) {}
+Heap<ValueType>::Heap() : size(0), height(0), root(nullptr) {}
+
+template<class ValueType>
+void Heap<ValueType>::insertValue(ValueType value) {
+    size++;
```

Comment:
Jak robisz inkrenetację a starcie to lepiej przy inicjalizacji zmiennej daj 1 a inkrementację przenieś do else, gdzie ponownie będziesz wchodzić do tej metody

Reply:
Dobra, ta niżej  to inna metoda. Robisz przeciążenie funkcji?

Reply:
Tak